src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Unix.cs

```diff
@@ -335,9 +335,21 @@ internal void LogBuffer(int size)
         private SocketError FinishOperationAccept(Internals.SocketAddress remoteSocketAddress)
         {
             System.Buffer.BlockCopy(_acceptBuffer!, 0, remoteSocketAddress.Buffer, 0, _acceptAddressBufferCount);
+            Socket? sukru = _acceptSocket;
             _acceptSocket = _currentSocket!.CreateAcceptSocket(
                 SocketPal.CreateSocket(_acceptedFileDescriptor),
                 _currentSocket._rightEndPoint!.Create(remoteSocketAddress));
+            if (sukru != null)
+            {
+                sukru.DisposeHandle();
+                sukru.CopyStateFromSource(_acceptSocket);
+                // We keep this socket to make clean-up.
+                Socket temp = _acceptSocket;
+                _acceptSocket = sukru;
+                temp.ClearHandle();
+                temp.Dispose();
```

Comment:
Do we have sufficient testing for disposing of a Socket that's had its handle nulled out?

Reply:
I'm not sure about it. I'll check it.